File: screenshot-fabrik.js
const puppeteer = require('puppeteer');
const fs = require('fs');

(async () => {
  const browser = await puppeteer.launch({
    headless: false,
    defaultViewport: {
      width: 1920,
      height: 1080
    }
  });
  
  const page = await browser.newPage();
  
  console.log('Navigating to page...');
  await page.goto('https://emilianoechevarriafever.github.io/fabrik-madrid-tables/', {
    waitUntil: 'networkidle2',
    timeout: 30000
  });
  
  // Wait a bit for any animations or lazy loading
  await page.waitForTimeout(2000);
  
  console.log('Taking initial screenshot...');
  await page.screenshot({
    path: 'fabrik-screenshot-1-full.png',
    fullPage: false
  });
  
  console.log('Taking full page screenshot...');
  await page.screenshot({
    path: 'fabrik-screenshot-2-fullpage.png',
    fullPage: true
  });
  
  // Scroll down to see ticket selector
  console.log('Scrolling to ticket selector...');
  await page.evaluate(() => {
    window.scrollBy(0, 600);
  });
  
  await page.waitForTimeout(1000);
  
  console.log('Taking ticket selector screenshot...');
  await page.screenshot({
    path: 'fabrik-screenshot-3-ticket-selector.png',
    fullPage: false
  });
  
  // Get information about the page state
  const pageInfo = await page.evaluate(() => {
    // Check which tab is selected
    const tabs = document.querySelectorAll('.nav-link, [role="tab"]');
    let selectedTab = 'None';
    tabs.forEach(tab => {
      if (tab.classList.contains('active')) {
        selectedTab = tab.textContent.trim();
      }
    });
    
    // Check for blue zones on the map
    const blueElements = [];
    const svgPaths = document.querySelectorAll('svg path, svg g, svg rect, svg polygon');
    svgPaths.forEach((el, idx) => {
      const fill = window.getComputedStyle(el).fill;
      const stroke = window.getComputedStyle(el).stroke;
      if (fill.includes('0, 0, 255') || fill.includes('rgb(0, 123, 255)') || 
          stroke.includes('0, 0, 255') || stroke.includes('rgb(0, 123, 255)') ||
          fill.includes('#007bff') || fill.includes('blue') || stroke.includes('blue')) {
        blueElements.push({
          id: el.id,
          class: el.className.baseVal || el.className,
          fill: fill,
          stroke: stroke
        });
      }
    });
    
    // Get ticket items
    const ticketItems = [];
    const tickets = document.querySelectorAll('.ticket-item, .list-group-item, [class*="ticket"]');
    tickets.forEach(ticket => {
      ticketItems.push(ticket.textContent.trim().substring(0, 100));
    });
    
    return {
      selectedTab,
      blueElementsCount: blueElements.length,
      blueElements: blueElements.slice(0, 10), // First 10
      ticketItemsCount: ticketItems.length,
      ticketItems: ticketItems.slice(0, 5) // First 5
    };
  });
  
  console.log('\n=== PAGE ANALYSIS ===');
  console.log('Selected Tab:', pageInfo.selectedTab);
  console.log('Blue Elements Found:', pageInfo.blueElementsCount);
  console.log('Blue Elements Details:', JSON.stringify(pageInfo.blueElements, null, 2));
  console.log('Ticket Items Found:', pageInfo.ticketItemsCount);
  console.log('Ticket Items:', pageInfo.ticketItems);
  
  fs.writeFileSync('fabrik-analysis.json', JSON.stringify(pageInfo, null, 2));
  
  console.log('\nScreenshots saved:');
  console.log('- fabrik-screenshot-1-full.png');
  console.log('- fabrik-screenshot-2-fullpage.png');
  console.log('- fabrik-screenshot-3-ticket-selector.png');
  console.log('- fabrik-analysis.json');
  
  await browser.close();
})();

File: screenshot-fabrik.py
#!/usr/bin/env python3
"""
Screenshot script for Fabrik Madrid tables page
Takes screenshots and analyzes the page structure
"""

import time
import json
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.chrome.options import Options

def main():
    # Setup Chrome options
    chrome_options = Options()
    chrome_options.add_argument('--window-size=1920,1080')
    # chrome_options.add_argument('--headless')  # Run in background
    
    print("Starting browser...")
    driver = webdriver.Chrome(options=chrome_options)
    
    try:
        url = 'https://emilianoechevarriafever.github.io/fabrik-madrid-tables/'
        print(f"Navigating to {url}...")
        driver.get(url)
        
        # Wait for page to load
        time.sleep(3)
        
        # Take initial screenshot
        print("Taking initial screenshot...")
        driver.save_screenshot('fabrik-screenshot-1-initial.png')
        
        # Analyze page state
        print("Analyzing page...")
        
        # Check selected tab
        selected_tab = "None"
        try:
            tabs = driver.find_elements(By.CSS_SELECTOR, '.nav-link, [role="tab"]')
            for tab in tabs:
                if 'active' in tab.get_attribute('class'):
                    selected_tab = tab.text.strip()
                    break
        except Exception as e:
            print(f"Error finding tabs: {e}")
        
        # Check for blue zones - look for SVG elements
        blue_zones = []
        try:
            svg_elements = driver.find_elements(By.CSS_SELECTOR, 'svg path, svg g[id], svg rect, svg polygon')
            for el in svg_elements:
                el_id = el.get_attribute('id') or ''
                el_class = el.get_attribute('class') or ''
                fill = el.value_of_css_property('fill')
                stroke = el.value_of_css_property('stroke')
                
                # Check if blue-ish color
                if any(color_check in str(fill).lower() or color_check in str(stroke).lower() 
                       for color_check in ['blue', '#007bff', 'rgb(0, 123, 255)', '007bff']):
                    blue_zones.append({
                        'id': el_id,
                        'class': el_class,
                        'fill': fill,
                        'stroke': stroke
                    })
        except Exception as e:
            print(f"Error finding blue zones: {e}")
        
        # Get ticket items
        ticket_items = []
        try:
            tickets = driver.find_elements(By.CSS_SELECTOR, '.ticket-item, .list-group-item, [class*="ticket"]')
            for ticket in tickets[:5]:  # First 5
                ticket_items.append(ticket.text.strip()[:100])
        except Exception as e:
            print(f"Error finding tickets: {e}")
        
        # Scroll down to ticket selector
        print("Scrolling to ticket selector...")
        driver.execute_script("window.scrollBy(0, 600);")
        time.sleep(2)
        
        print("Taking ticket selector screenshot...")
        driver.save_screenshot('fabrik-screenshot-2-tickets.png')
        
        # Scroll to top
        driver.execute_script("window.scrollTo(0, 0);")
        time.sleep(1)
        
        # Take full page screenshot
        print("Taking full page screenshot...")
        # Get full page height
        total_height = driver.execute_script("return document.body.scrollHeight")
        driver.set_window_size(1920, total_height)
        time.sleep(1)
        driver.save_screenshot('fabrik-screenshot-3-fullpage.png')
        
        # Prepare analysis
        analysis = {
            'selected_tab': selected_tab,
            'blue_zones_count': len(blue_zones),
            'blue_zones': blue_zones[:10],  # First 10
            'ticket_items_count': len(ticket_items),
            'ticket_items': ticket_items,
            'page_url': url,
            'timestamp': time.strftime('%Y-%m-%d %H:%M:%S')
        }
        
        # Save analysis
        with open('fabrik-analysis.json', 'w') as f:
            json.dump(analysis, f, indent=2)
        
        print("\n=== PAGE ANALYSIS ===")
        print(f"Selected Tab: {selected_tab}")
        print(f"Blue Zones Found: {len(blue_zones)}")
        if blue_zones:
            print("Blue Zones Details:")
            for zone in blue_zones[:5]:
                print(f"  - ID: {zone['id']}, Class: {zone['class']}")
        print(f"Ticket Items Found: {len(ticket_items)}")
        if ticket_items:
            print("Ticket Items:")
            for item in ticket_items:
                print(f"  - {item[:80]}")
        
        print("\nScreenshots saved:")
        print("- fabrik-screenshot-1-initial.png")
        print("- fabrik-screenshot-2-tickets.png")
        print("- fabrik-screenshot-3-fullpage.png")
        print("- fabrik-analysis.json")
        
    finally:
        print("\nClosing browser...")
        driver.quit()

if __name__ == '__main__':
    main()
